### PR TITLE
Round-trip system printing

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/ModelingToolkit.jl [hunk 1]
```diff
@@ -154,6 +154,7 @@ export ode_order_lowering, liouville_transform
 export runge_kutta_discretize
 export PDESystem
 export Reaction, ReactionSystem, ismassaction, oderatelaw, jumpratelaw
+export system2expr
```
Comment (ChrisRackauckas) [inline src/ModelingToolkit.jl:157]: ehh, too cute of a name? system_to_expr? `toexpr(::System)`?

File: src/systems/abstractsystem.jl [hunk 2]
```diff
@@ -411,6 +411,98 @@ function (f::AbstractSysToExpr)(O)
...
+    var2name = Dict{Any,Symbol}()
+    for v in Iterators.flatten((sts, ps))
+        var2name[v] = getname(v)
+    end
+
+    eqs_name = push_eqs!(stmt, equations(sys), var2name)
+    defs_name = push_defaults!(stmt, defaults(sys), var2name)
+
+    if sys isa ODESystem
+        push!(stmt, :($ODESystem($eqs_name, $ivname, $stsname, $psname; defaults=$defs_name)))
```
Comment (ChrisRackauckas) [inline src/systems/abstractsystem.jl:498]: seems like this machinery needs to dispatch per system?

File: src/ModelingToolkit.jl (hunk 1 above)
Comment (YingboMa) [reply src/ModelingToolkit.jl:157]: Yeah, overloading `toexpr` is better.

File: src/systems/abstractsystem.jl (hunk 2 above)
Comment (YingboMa) [reply src/systems/abstractsystem.jl:498]: Yeah, I plan to do that when I clean up the code.

**Comment by ChrisRackauckas** [comment]
Needs a test, but it does look good to go with a name change. And yeah, it's going to need to be refactored to each system type, etc., but that can come later.

**Comment by YingboMa** [comment]
There are tests for ODESystem and NonlinearSystem.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
oh missed that.

**Pull request merged**